File: python-requests/result.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import json
import requests
import sys
from conf import appid, apikey, sender, receivers, content

if len(sys.argv) == 1:
    print 'Usage:\n python %s deliveryid' % sys.argv[0]
    sys.exit(1)

deliveryid = sys.argv[1]

url = 'https://api.bluehouselab.com/smscenter/v1.0/result/'+deliveryid
params = {
    'sender'     : sender,
    'receivers'  : receivers,
    'content'    : content,
}
headers = {'Content-type': 'application/json; charset=utf-8',}
r = requests.get(url, auth=(appid, apikey), headers=headers)

print r.status_code, r.reason
if r.status_code == 200:
    print r.json()

File: node.js/result.js
var c = require('./conf');
var https = require("https");
var credential = 'Basic '+new Buffer(c.APPID+':'+c.APIKEY).toString('base64');
if(process.argv.length != 3) {
	console.log('Usage: ');
	console.log(' node result.js delivery-id');
	process.exit(1);
}
var deliveryid = process.argv[2];

var options = {
  host: 'api.bluehouselab.com',
  port: 443,
  path: '/smscenter/v1.0/result/'+deliveryid,
  headers: {
    'Authorization': credential,
  },
  method: 'GET'
};
var req = https.request(options, function(res) {
  console.log(res.statusCode);
  var body = "";
  res.on('data', function(d) {
    body += d;
  });
  res.on('end', function(d) {
  	if(res.statusCode==200)
		console.log(JSON.parse(body));
	else
		console.log(body);
  });
});
req.end();
req.on('error', function(e) {
	console.error(e);
});
